service/history/outbound_queue_standby_task_executor.go

```diff
@@ -75,14 +93,152 @@ func (e *outboundQueueStandbyTaskExecutor) Execute(
 		metrics.OperationTag(taskType),
 	}
 
-	if err == nil {
-		smRegistry := e.shardContext.StateMachineRegistry()
-		err = smRegistry.ExecuteStandbyImmediateTask(ctx, e, ref, smt)
+	if err != nil {
+		return queues.ExecuteResponse{
+			ExecutionMetricTags: metricsTags,
+			ExecutedAsActive:    false,
+			ExecutionErr:        err,
+		}
+	}
+
+	actionFn := func(ctx context.Context) (any, error) {
+		err := e.Access(ctx, ref, hsm.AccessRead, func(node *hsm.Node) error {
+			if smt.Concurrent() {
+				//nolint:revive // concurrent tasks implements hsm.ConcurrentTask interface
+				concurrentSmt := smt.(hsm.ConcurrentTask)
+				return concurrentSmt.Validate(node)
+			}
+			return nil
+		})
+		return nil, err
 	}
 
+	err = e.processTask(
+		ctx,
+		task,
+		actionFn,
+		getStandbyPostActionFn(
+			task,
+			e.getCurrentTime,
+			dynamicconfig.GetDurationPropertyFnFilteredByTaskType(2*time.Minute)(task.GetType()),
+			dynamicconfig.GetDurationPropertyFnFilteredByTaskType(5*time.Minute)(task.GetType()),
+			e.fetchHistoryFromRemote,
+			standbyTimerTaskPostActionTaskDiscarded,
+		),
+	)
+
 	return queues.ExecuteResponse{
 		ExecutionMetricTags: metricsTags,
 		ExecutedAsActive:    false,
 		ExecutionErr:        err,
 	}
 }
+
+func (e *outboundQueueStandbyTaskExecutor) processTask(
+	ctx context.Context,
+	task tasks.Task,
+	actionFn func(context.Context) (any, error),
+	postActionFn standbyPostActionFn,
+) (retError error) {
+	ctx, cancel := context.WithTimeout(ctx, taskTimeout)
+	defer cancel()
+
+	nsRecord, err := e.shardContext.GetNamespaceRegistry().GetNamespaceByID(
+		namespace.ID(task.GetNamespaceID()),
+	)
+	if err != nil {
+		return err
+	}
+	if !nsRecord.IsOnCluster(e.clusterName) {
+		// namespace is not replicated to local cluster, ignore corresponding tasks
+		return nil
+	}
+
+	historyResendInfo, err := actionFn(ctx)
+	if err != nil {
+		return err
+	}
+
+	return postActionFn(ctx, task, historyResendInfo, e.logger)
+}
+
+func (e *outboundQueueStandbyTaskExecutor) fetchHistoryFromRemote(
```

Comment:
I didn't review this, I'm assuming it's copied from the other standby executors.

One thing to note here is that fetching history may not help here since some state machines transition without generating history events. @yycptt please account for this in the long term state replication work.

Reply:
Btw, noticed an issue with this PR as it is. The `actionFn` needs to return a `historyResendInfo` for the `fetchHistoryFromRemote` to work, and to build the `historyResendInfo` it needs the mutable state, which is not available in the context of `actionFn` here. We might need to replace the `stateMachineEnvironment.Access` with manually building the validation of stale task to have the mutable state, or change the `stateMachineEnvironment.Access` to call `accessor` with mutable state.
cc: @bergundy